src/proxy/processors/push-action/parsePush.js

```diff
@@ -24,41 +25,96 @@ const exec = async (req, action) => {
     action.commitData = getCommitData(contents);
 
     if (action.commitFrom === '0000000000000000000000000000000000000000') {
-      action.commitFrom =
-        action.commitData[action.commitData.length - 1].parent;
+      action.commitFrom = action.commitData[action.commitData.length - 1].parent;
     }
 
+    const user = action.commitData[action.commitData.length - 1].committer;
+    console.log(`Push Request recevied from user ${user}`);
+    action.user = user;
+
     step.content = {
       meta: meta,
     };
   } catch (e) {
-    step.setError(e.toString('utf-8'));
-    throw e;
+    step.setError(
+      `Unable to parse push. Please contact an administrator for support: ${e.toString('utf-8')}`,
+    );
   } finally {
     action.addStep(step);
-    return action;
   }
+  return action;
 };
 
 const getCommitData = (contents) => {
+  console.log({ contents });
   return lod
     .chain(contents)
     .filter({ type: 1 })
     .map((x) => {
-      const parts = x.content.split('\n');
-      const tree = parts[0];
-      const parent = parts[1];
-      const author = parts[2];
-      const committer = parts[3];
-      const message = parts[5];
+      console.log({ x });
+
+      const formattedContent = x.content.split('\n');
+      console.log({ formattedContent });
+
+      const parts = formattedContent.filter((part) => part.length > 0);
+      console.log({ parts });
+
+      const tree = parts
+        .find((t) => t.split(' ')[0] === 'tree')
+        .replace('tree', '')
+        .trim();
+      console.log({ tree });
+
+      const parent = parts
+        .find((t) => t.split(' ')[0] === 'parent')
+        .replace('parent', '')
+        .trim();
+      console.log({ parent });
+
+      const author = parts
+        .find((t) => t.split(' ')[0] === 'author')
+        .replace('author', '')
+        .trim();
+      console.log({ author });
+
+      const committer = parts
+        .find((t) => t.split(' ')[0] === 'committer')
+        .replace('committer', '')
+        .trim();
+      console.log({ committer });
+
+      const indexOfMessages = formattedContent.indexOf('');
+      console.log({ indexOfMessages });
+
+      const message = formattedContent
+        .slice(indexOfMessages + 1, formattedContent.length - 1)
+        .join(' ');
+      console.log({ message });
+
+      const commitTimestamp = committer.split(' ')[2];
+      console.log({ commitTimestamp });
+
+      const authorEmail = author.split(' ')[1].slice(1, -1);
+      console.log({ authorEmail });
```

Comment:
Some of the parsing logic seems bugged. From the logs:

```
[0] { author: 'Thomas Cooper <coopernetes@proton.me> 1705253171 -0500' }
[0] { committer: 'Thomas Cooper <coopernetes@proton.me> 1705253171 -0500' }
[0] { indexOfMessages: 11 }
[0] { message: 'test' }
[0] { commitTimestamp: '<coopernetes@proton.me>' }
[0] { authorEmail: 'oope' }
```

```suggestion
      const authorEmail = author.substring(author.indexOf('<')+1, author.indexOf('>'));
      console.log({ authorEmail });
```

Reply:
@coopernetes - could you suggest a solution which encapsulates both a space-separated name and GitHub username at the zero-index? I think we agreed last time this is likely what is causing the issue of indexing for the rest of the properties.